draft-ietf-ppm-dap.md

```diff
@@ -575,24 +573,29 @@ particularly important for controlling the amount of noise added to reports
 by Clients (or added to aggregate shares by Aggregators) in order to achieve
 differential privacy.
 
-To get the aggregate of the next batch, the Collector issues a query of type
-`next-batch`. (See FixedSizeQuery in {{query}}.) The response to this query (see
-{{collect-flow}}) includes a "batch ID", which can be used to query the same
-batch later on. This is done using the `by-batch-id` batch-query type.
-
 For this query type, the Aggregators group reports into arbitrary batches such
-that each batch has roughly the same number of reports. In addition to the
-minumum batch size common to all query types, the conifugration includes a
-"maximum batch size", `max_batch_size`, that determines maximum number of
-reports per batch.
+that each batch has roughly the same number of reports. These batches are
+identified by "batch IDs", which are numerical identifiers allocated
+consecutively starting from 0.
+
+To get the aggregate of a batch, the Collector issues a query of type
+`by-batch-id`, specifying the batch ID of interest in the query. (See
+`FixedSizeQuery` in {{query}}.) The Collector may not know which batch ID it is
+interested in (for example, in startup scenarios); in this case, it can also
```

Comment:
The `current-batch` functionality may not be ideal:

* It assumes that a Collector might want to start up against a task that has already collected some batches, and will want to start from "recent" data. If this is not the case (for example, if we assume that a Collector can always safely start from batch ID 0, or are OK saying that communicating a "recent" batch ID can be done out-of-band), we could drop this functionality entirely.
* Even if we keep this functionality, it would be nicer to separate "give me a recent batch ID" from "start collecting a batch ID". One way to do this would be to expose an endpoint that forwards to a recent batch ID; this would be easier to specify with #278, at which point the batch ID might be specified as part of the URL.

Reply:
Yeah, this is a bit hairy. The protocol seems like it would be a lot simpler if all the Collector could do is specify which batch ID it wants.

Do you thin this functionality is needed right now? If not, maybe better to punt on and keep DAP-02 simple.

Reply:
I think an eventual "final"/"production-quality" design would definitely want some provision to allow Collectors to discover a recent batch ID from the Aggregators.

However, you're probably right that this isn't needed for initial testing: I suspect interop-test-level deployments can get away with the Collector always starting at batch ID 0, and if that doesn't work out, out-of-band communications would allow configuration of the Collector to start from the correct batch ID.

I went ahead and tore out the `current-batch` functionality for now.

Reply:
Sounds good. Most likely we'll be able to spell this feature when we actually need it.